src/mediarithmics/plugins/ad-renderer/base/AdRendererBasePlugin.ts

```diff
@@ -101,6 +101,17 @@ export abstract class AdRendererBasePlugin<T extends AdRendererBaseInstanceConte
     instanceContext: T
   ): Promise<AdRendererPluginResponse>;
 
+  protected async getInstanceContext(creativeId: string, forceReload: boolean): Promise<T> {
+    if (!this.pluginCache.get(creativeId) || forceReload){
+      this.pluginCache.put(
+        creativeId,
+        this.instanceContextBuilder(creativeId, forceReload).catch(() => this.pluginCache.del(creativeId)),
+        this.getInstanceContextCacheExpiration()
+      )
+    }
+    return Promise.resolve(this.pluginCache.get(creativeId))
```

Comment:
`pluginCache` already contains a promise, you can remove the extra `Promise.resolve()` (same for others).